lib/jxl/encode.cc

```diff
@@ -607,7 +607,11 @@ JxlEncoderStatus JxlEncoderAddJPEGFrame(const JxlEncoderOptions* options,
 JxlEncoderStatus JxlEncoderAddImageFrame(const JxlEncoderOptions* options,
                                          const JxlPixelFormat* pixel_format,
                                          const void* buffer, size_t size) {
-  if (!options->enc->basic_info_set || !options->enc->color_encoding_set) {
+  if (!options->enc->basic_info_set ||
+    !options->enc->color_encoding_set && !options->enc->metadata.m.xyb_encoded) {
```

Comment:
adding parentheses here for clarify will fix the "'&&' within '||'" we see in some tests.

Reply:
I thought it was decently clear since I had a newline, but I'll add parentheses anyway just to make it extra clear.